src/centralstore/incremental/RelationBlock.cpp

```diff
@@ -0,0 +1,332 @@
+                                          RECORD_SIZE)) {  // < ------ relation data offset ID = 9
+        relation_block_logger.error(
+            "ERROR: Error while reading relation destination previous relation partition id data offset ID = 9 from "
+            "relation block address " +
+            std::to_string(address));
+        return NULL;
+    }
+
+    if (!RelationBlock::relationsDB->read(reinterpret_cast<char*>(&propertyReference),
+                                          RECORD_SIZE)) {  // < ------ relation data offset ID = 10
+        relation_block_logger.error(
+            "ERROR: Error while reading relation property address data offset ID = 10 from relation block address " +
+            std::to_string(address));
+        return NULL;
+    }
+
+    return new RelationBlock(address, source, destination, propertyReference);
+}
+
+RelationBlock* RelationBlock::nextSource() { return RelationBlock::get(this->source.nextRelationId); }
+
+RelationBlock* RelationBlock::previousSource() { return RelationBlock::get(this->source.preRelationId); }
+
+RelationBlock* RelationBlock::nextDestination() { return RelationBlock::get(this->destination.nextRelationId); }
+
+RelationBlock* RelationBlock::previousDestination() { return RelationBlock::get(this->destination.preRelationId); }
+
+bool RelationBlock::setNextSource(unsigned int newAddress) {
+    if (this->updateRelationRecords(2, newAddress)) {
+        this->source.nextRelationId = newAddress;
+    } else {
+        throw "Exception: Error while updating the relation next source address " + std::to_string(newAddress);
+    }
+    return true;
+}
+
+bool RelationBlock::setPreviousSource(unsigned int newAddress) {
+    if (this->updateRelationRecords(4, newAddress)) {
+        this->source.preRelationId = newAddress;
+
+    } else {
+        throw "Exception: Error while updating the relation previous source address " + std::to_string(newAddress);
+    }
+    return true;
+}
+
+bool RelationBlock::setNextDestination(unsigned int newAddress) {
+    if (this->updateRelationRecords(6, newAddress)) {
+        this->destination.nextRelationId = newAddress;
+    } else {
+        throw "Exception: Error while updating the relation next destination address " + std::to_string(newAddress);
+    }
+    return true;
+}
+
+bool RelationBlock::setPreviousDestination(unsigned int newAddress) {
+    if (this->updateRelationRecords(8, newAddress)) {
+        this->destination.preRelationId = newAddress;
+    } else {
+        throw "Exception: Error while updating the relation previous destination address " + std::to_string(newAddress);
+    }
+    return true;
+}
+
+/**
+ * Update relation record block given the offset to the recored from the begining, i:e
+ *  recordOffset 0 --> Source address
+ *  recordOffset 1 --> Source address
+ *  recordOffset 2 --> Source's next relation block address
+ *  recordOffset 3 --> Source's next relation block partition id
```

Comment:
Better if we could document the rest of the recordOffset values (4, 5, ...,9) as well here

Reply:
Fixed this in https://github.com/miyurud/jasminegraph/pull/115/commits/3ca3404f663a5f12c3ce466721b8fc435447cd26

Reply:
Done